src/hotspot/cpu/aarch64/aarch64_neon.ad

```diff
@@ -4243,32 +4243,64 @@ instruct vxor16B(vecX dst, vecX src1, vecX src2)
 
 // ------------------------------ Shift ---------------------------------------
 
-instruct vshiftcnt8B(vecD dst, iRegIorL2I cnt) %{
+// Vector shift count
+// Note-1: Low 8 bits of each element are used, so it doesn't matter if we
+//         treat it as ints or bytes here.
+// Note-2: Shift value is negated for RShiftCntV additionally. See the comments
+//         on vsra8B rule for more details.
+
+instruct vslcnt8B(vecD dst, iRegIorL2I cnt) %{
   predicate(n->as_Vector()->length_in_bytes() == 4 ||
-            n->as_Vector()->length_in_bytes() == 8);
+                            n->as_Vector()->length_in_bytes() == 8);
   match(Set dst (LShiftCntV cnt));
-  match(Set dst (RShiftCntV cnt));
+  ins_cost(INSN_COST);
   format %{ "dup  $dst, $cnt\t# shift count vector (8B)" %}
   ins_encode %{
     __ dup(as_FloatRegister($dst$$reg), __ T8B, as_Register($cnt$$reg));
   %}
   ins_pipe(vdup_reg_reg64);
 %}
 
-instruct vshiftcnt16B(vecX dst, iRegIorL2I cnt) %{
+instruct vslcnt16B(vecX dst, iRegIorL2I cnt) %{
   predicate(n->as_Vector()->length_in_bytes() == 16);
   match(Set dst (LShiftCntV cnt));
-  match(Set dst (RShiftCntV cnt));
+  ins_cost(INSN_COST);
   format %{ "dup  $dst, $cnt\t# shift count vector (16B)" %}
   ins_encode %{
     __ dup(as_FloatRegister($dst$$reg), __ T16B, as_Register($cnt$$reg));
   %}
   ins_pipe(vdup_reg_reg128);
 %}
 
+instruct vsrcnt8B(vecD dst, iRegIorL2I cnt) %{
+  predicate(UseSVE == 0 && (n->as_Vector()->length_in_bytes() == 4 ||
```

Comment:
As mentioned in the commit msg, I think `UseSVE == 0` check is not needed here.

Reply:
Again, the place to fix that is JDK head.

Reply:
But `UseSVE == 0` check is removed for `vslcnt8B` and `vslcnt16B` already.
As stated in the commit msg, VectorAPI  SVE codegen part is missing for 17u.

Reply:
> But `UseSVE == 0` check is removed for `vslcnt8B` and `vslcnt16B` already. As stated in the commit msg, VectorAPI SVE codegen part is missing for 17u.

OK.

Reply:
Thanks, I removed 2 checks symmetrically.